biolink/pieces-parts/rank-regulators-exe.rkt

```diff
@@ -0,0 +1,127 @@
+                       genes (rG) that the regulator gene (G) regulates along with additional information.
+                       Each element in the value is a list that contains the rG's curie, name, the presumed
+                       direction of regulation (told by predicates), and provenance for the edges in the form
+                       of pubmed ids. The functions 1-hop-query/report and find-intersections also include
+                       a boolean that indicates whether or not the G and rG directly interact with each other
+                       in the body (#t if yes) and provenance for that information also in the form of pubmed
+                       ids.
+
+Note: The functions called in this file are more extensively documented in rank-regulators.rkt
+|#
+
+#|
+The following calls are an example based off an ARDS gene list, defined as ards-genes. There are 517 ENSEMBL
+gene curies on that list, and they yield 468 uniprot curies.
+
+Note: runtimes given in comments below are based off a 2020 MacBook Pro (2 GHz Quad-Core Intel Core i5,
+      16G RAM). They are meant to give an idea of which calls take a lot longer to finish.
+|#
+
+#|
+Turns the inputted gene list of interest into a list of uniprot curies in order to be passed
+into the GO query.
+runtime: < 30s
+|#
+;;function call: genes->uniprots | takes: list of gene curies | returns: list of uniprot curies
+(define uniprots (time (genes->uniprots ards-genes)))
+
+#|
+The next two statements 1) run the GO query and 2) reports the results of the query in an association list.
+runtime:
+    GO-query-edges: ~5 min
+    go-query-results: just under 2 hrs
+|#
+;;function call: query-GO | takes: list of uniprot curies | returns: list of three edge hashes
+(define GO-query-edges (time (query-GO uniprots)))
+
+;;function call: report-GO-queries | takes: list of three edge hashes | returns: sorted assoc list
+(define go-query-results (time (report-GO-queries GO-query-edges)))
+
+#|
+Runs and reports the 1-hop query.
+runtime: ~ 20 min
+|#
+;;function call: 1-hop-query/report | takes: list of uniprot curies | returns: sorted assoc list
+(define 1-hop-results (time (1-hop-query/report uniprots)))
+
+#|
+Finds the intersections between the GO query findings and 1-hop query findings.
+|#
+;;function call: find-intersections | takes: sorted assoc list returned by report-GO-queries, sorted
+;;assoc list returned by 1-hop-query/report | returns: sorted assoc list
+;; NOTE: CURRENTLY THROWS ERROR
+;;(define intersections (time (find-intersections go-query-results 1-hop-results)))
+
+;;function call: find-expression-locations | takes: list of gene/protein curies | returns: hash that maps cell/tissue type to list of gene/proteins
+;;runtime: ~25 min
+(define cell/tissue-types (time (find-expression-locations (set-union (map caar go-query-results) (map caar 1-hop-results) uniprots))))
```

Comment:
the timing gets triggered over once per second. i suspect this slows down the actual computation.